File: backend/model/Transaction.js
const mongoose = require('mongoose')
const Schema = mongoose.Schema

const transactionSchema = new Schema(
  {
    userId: {
      type: Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    labelId: {
      type: Schema.Types.ObjectId,
      ref: 'Label',
      required: true,
    },
    transactionType: {
      type: String,
      enum: ['income', 'expense'],
      required: true,
    },
    amount: {
      type: Number,
      required: true,
    },
    description: {
      type: String,
      required: true,
    },
    date: {
      type: Date,
      required: true,
    },
  },
  {
    timestamps: true,
  }
)

module.exports = mongoose.model('Transaction', transactionSchema)

File: frontend/lib/model/transaction.dart
class Transaction {
  final String id;
  final String labelId;
  final String transactionType;
  final num amount;
  final String description;
  final String date;

  Transaction({
    required this.id,
    required this.labelId,
    required this.transactionType,
    required this.amount,
    required this.description,
    required this.date,
  });

  factory Transaction.fromJson(Map<String, dynamic> json) {
    return Transaction(
      id: json['_id'],
      labelId: json['labelId'],
      transactionType: json['transactionType'],
      amount: json['amount'].toDouble(),
      description: json['description'],
      date: json['date'],
    );
  }
}
